src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -500,6 +505,179 @@ subroutine gsi_rfv3io_get_grid_specs(ierr)
+    iret=nf90_inquire(gfile_grid_spec,ndimensions,nvariables,nattributes,unlimiteddimid)
+    gfile_loc=gfile_grid_spec
+    do k=1,ndimensions
+       iret=nf90_inquire_dimension(gfile_loc,k,name,len)
+       if(trim(name)=='grid_xt') nxens=len
+       if(trim(name)=='grid_yt') nyens=len
+    enddo
+    allocate(grid_lat(nxens+1,nyens+1))
+    allocate(grid_lon(nxens+1,nyens+1))
+    allocate(grid_latt(nxens,nyens))
+    allocate(grid_lont(nxens,nyens))
+    do k=ndimensions+1,nvariables
+       iret=nf90_inquire_variable(gfile_loc,k,name,len)
+       if(trim(name)=='grid_lat') then
+          iret=nf90_get_var(gfile_loc,k,grid_lat)
+       endif
+       if(trim(name)=='grid_lon') then
+          iret=nf90_get_var(gfile_loc,k,grid_lon)
+       endif
+       if(trim(name)=='grid_latt') then
+          iret=nf90_get_var(gfile_loc,k,grid_latt)
+       endif
+       if(trim(name)=='grid_lont') then
+          iret=nf90_get_var(gfile_loc,k,grid_lont)
+       endif
+    enddo
+    iret=nf90_close(gfile_loc)
+
+    nlon_regionalens=nxens
+    nlat_regionalens=nyens
+    allocate(ny_layout_lenens(0:fv3_io_layout_y-1))
+    allocate(ny_layout_bens(0:fv3_io_layout_y-1))
+    allocate(ny_layout_eens(0:fv3_io_layout_y-1))
+    ny_layout_lenens=nyens
+    ny_layout_bens=0
+    ny_layout_eens=0
+    if(fv3_io_layout_y > 1) then
+       allocate(gfile_loc_layout(0:fv3_io_layout_y-1))
+       do nio=0,fv3_io_layout_y-1
+          write(filename_layout,'(a,a,I4.4)') trim(grid_spec),'.',nio
+          iret=nf90_open(filename_layout,nf90_nowrite,gfile_loc_layout(nio))
+          if(iret/=nf90_noerr) then
+             write(6,*)' problem opening ',trim(filename_layout),', Status =',iret
+             ierr=1
+             return
+          endif
+          iret=nf90_inquire(gfile_loc_layout(nio),ndimensions,nvariables,nattributes,unlimiteddimid)
+          do k=1,ndimensions
+              iret=nf90_inquire_dimension(gfile_loc_layout(nio),k,name,len)
+              if(trim(name)=='grid_yt') ny_layout_lenens(nio)=len
+          enddo
+          iret=nf90_close(gfile_loc_layout(nio))
+       enddo
+       deallocate(gfile_loc_layout)
+! figure out begin and end of each subdomain restart file
+       nylen=0
+       do nio=0,fv3_io_layout_y-1
+          ny_layout_bens(nio)=nylen + 1
+          nylen=nylen+ny_layout_lenens(nio)
+          ny_layout_eens(nio)=nylen
+       enddo
+    endif
+    if(mype==0)write(6,*),'nxens,nyens=',nxens,nyens
+    if(mype==0)write(6,*),'ny_layout_lenens=',ny_layout_lenens
+    if(mype==0)write(6,*),'ny_layout_bens=',ny_layout_bens
+    if(mype==0)write(6,*),'ny_layout_eens=',ny_layout_eens
+
+!
+!  need to decide the grid orientation of the FV regional model   
+!
+!   grid_type_fv3_regional = 0 : decide grid orientation based on
+!                                grid_lat/grid_lon
+!                            1 : input is E-W N-S grid
+!                            2 : input is W-E S-N grid
+!
+    if(grid_type_fv3_regional == 0) then
```

Comment:
Since dual_res is to deal with ensemble forecast being different  from the control forecast, will it be necessary to consider if they are also different in the grid orientation?  Or, at least, do a test to make sure they are are same ( only one grid_type_fv3_regional parameter is needed).